### PR TITLE
Bookstore: initialize project with components

### PR DESCRIPTION
**Pull request by Uzair-Manzoor**
## This pull request has been created against the below mentioned implementations into the project:

- Created a new React application using [Create React App](https://reactjs.org/docs/create-a-new-react-app.html#create-react-app)
- Created a directory for your reusable components: `/components`.
- Installed [React Router V6](https://reactrouter.com/en/main/start/tutorial#setup)
- Added 2 routes in the application, each rendering a page component:
  - `/`
  - `/categories`
- Created components inside of the `/components` directory

### CODE REVIEW

File: src/components/BookPage.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    { id: uuidv4(), title: 'Moby Dick', author: 'Herman Melville' },
+    { id: uuidv4(), title: 'War and Peace', author: 'Leo Tolstoy' },
+    { id: uuidv4(), title: 'The Brothers Karamazov', author: 'Fyodor Dostoyevsky' },
+  ];
+
+  return (
+    <>
+      <ul className="books">
+        {books.map((book) => (
+          <Book key={book.id} title={book.title} author={book.author} />
```
Comment (Johnadibe) [inline src/components/BookPage.js:17]: - [ ] According to the project requirements, you should create a component that contains a list to render individual books.  I recommend creating a separate component named `BookList `specifically for this purpose. Doing so will help you meet the project requirements more effectively.👍

book

You could use the below as a guide:
```
// In Book.js
import React, { useState } from 'react';

const Book = ({ book, onDelete }) => {
  const [isDeleted, setIsDeleted] = useState(false);

  const handleDelete = () => {
    onDelete(book.id);
    setIsDeleted(true);
  };

  return (
    <div className={`book ${isDeleted ? 'deleted' : ''}`}>
      <h3>{book.title}</h3>
      <p>Author: {book.author}</p>
      <button onClick={handleDelete}>Delete</button>
    </div>
  );
};


```

```
// In BookList.js
import Book from './Book';

const BookList = ({ books, onDelete }) => {
  return (
    <div className="book-list">
      {books.map((book) => (
        <Book key={book.id} book={book} onDelete={onDelete} />
      ))}
    </div>
  );
};


```

```
// In BookPage.js
import BookList from './components/BookList';

function BookPage() {
  const books = [
    { id: 1, title: 'Book 1', author: 'Author 1' },
    { id: 2, title: 'Book 2', author: 'Author 2' },
    { id: 3, title: 'Book 3', author: 'Author 3' },
  ];

  const handleDeleteBook = (id) => {
    const updatedBooks = books.filter((book) => book.id !== id);
    setBooks(updatedBooks);
  };

  return (
    <>
      <BookList books={books} onDelete={handleDeleteBook} />
    </>
  );
}

```

**Review by Johnadibe — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES ♻️
### Hi @Uzair-Manzoor👋🏽, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

mario_ladders

## Highlights
✅ You used GitFlow correctly 👍🏽
✅ Linter checks passed✔️
✅ Added 2 routes in the application in which each renders a page component✔
✅ Created a component inside of the /components directory that contains individual book state as well as a button to delete the book✔
✅ Created a component inside of the /components directory that contains a form to create new books✔
✅ Created a component inside of the /components directory that contains a navigation to link to the 2 routes✔


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/BookPage.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    { id: uuidv4(), title: 'The Great Gatsby', author: 'Scott Fitzgerald' },
+    { id: uuidv4(), title: 'Moby Dick', author: 'Herman Melville' },
+    { id: uuidv4(), title: 'War and Peace', author: 'Leo Tolstoy' },
+    { id: uuidv4(), title: 'The Brothers Karamazov', author: 'Fyodor Dostoyevsky' },
+  ];
+
+  return (
+    <>
+      {/* rander */}
+      <Book key={books.id} title={books.title} author={books.author} BookList books={books} />
```
Comment (bienvenuushindi) [inline src/components/BookPage.js]: :red_circle: [REQUIRED]

1. You've correctly defined the BookList component to accept a books prop, but in the BookPage component, you should pass the books array to BookList like this:

`<BookList books={books} />`
This way, the BookList component can access the list of books you've defined.

File: src/components/BookList.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const BookList = ({ books }) => (
+  <ul className="books">
+    {books.map((book) => (
+      <books key={book.id} title={book.title} />
```
Comment (bienvenuushindi) [inline src/components/BookList.js]: :red_circle: [REQUIRED]
In the this component, when mapping through the books array, use the Book component to render each book item:
```
{books.map((book) => (
  <Book key={book.id} title={book.title} author={book.author} />
))}

```

File: src/components/book.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import PropTypes from 'prop-types';
+
+const Book = (props) => {
+  const { title, author } = props;
```
Comment (bienvenuushindi) [inline src/components/book.js:4]: :orange_circle: [OPTIONAL]
The Book component should receive its props directly as function arguments, like this:
```
const Book = ({ title, author }) => {
  // Your component code here
}
```

**Review by bienvenuushindi — CHANGES_REQUESTED** [review]
Hi @Uzair-Manzoor  , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
May I highlight the following:

- The repo looks professional.:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:
- Your project looks good:heavy_check_mark:

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by V-Blaze — APPROVED** [review]
Hi @Uzair-Manzoor  , 

# Approved 🟢 🟢 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy


## Highlights

- App initialized using `Create React App`💯
-  Routing set up correctly 👍
- App runs as expected🟢 👍

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**